packages/contract/src/test-utils/launch-test-node.ts

```diff
@@ -0,0 +1,204 @@
+import type { JsonAbi } from '@fuel-ts/abi-coder';
+import type { WalletUnlocked } from '@fuel-ts/account';
+import { setupTestProviderAndWallets } from '@fuel-ts/account/test-utils';
+import type {
+  LaunchCustomProviderAndGetWalletsOptions,
+  SetupTestProviderAndWalletsReturn,
+} from '@fuel-ts/account/test-utils';
+import { FuelError } from '@fuel-ts/errors';
+import type { Contract } from '@fuel-ts/program';
+import type { SnapshotConfigs } from '@fuel-ts/utils';
+import { getForcProject } from '@fuel-ts/utils/test-utils';
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import { mergeDeepRight } from 'ramda';
+
+import type { DeployContractOptions } from '../contract-factory';
+import ContractFactory from '../contract-factory';
+
+interface DeployContractConfig {
+  /**
+   * Directory of contract to be deployed. The sway program **must** be built beforehand.
+   */
+  contractDir: string;
+  /**
+   * Index of wallet to be used for deployment. Defaults to `0` (first wallet).
+   */
+  walletIndex?: number;
+  /**
+   * Options for contract deployment taken from `ContractFactory`.
+   */
+  options?: DeployContractOptions;
+  /**
+   * Used to specify the folder where the built binary is located.
+   * Default is `release`, which corresponds  to `contractDir/out/release`.
+   */
+  build?: Parameters<typeof getForcProject>[0]['build'];
+}
+
+interface LaunchTestNodeOptions extends LaunchCustomProviderAndGetWalletsOptions {
+  /**
+   * Pass in either the path to the contract's root directory to deploy the contract or use `DeployContractConfig` for more control.
+   */
+  deployContracts: (DeployContractConfig | string)[];
+}
+
+interface LaunchTestNodeReturn<TContracts> extends SetupTestProviderAndWalletsReturn {
+  contracts: TContracts;
+}
+function getChainSnapshot(nodeOptions: LaunchTestNodeOptions['nodeOptions']) {
+  let envChainMetadata: SnapshotConfigs['metadata'] | undefined;
```

Comment:
I think we want to handle errors from `readFileSync`

```typescript
let envChainMetadata: SnapshotConfigs['metadata'] | undefined;
let chainConfig: SnapshotConfigs['chainConfig'] | undefined;
let stateConfig: SnapshotConfigs['stateConfig'] | undefined;

const defaultChainSnapshotDir = process.env.DEFAULT_CHAIN_SNAPSHOT_DIR;

if (defaultChainSnapshotDir) {
  envChainMetadata = readJsonFile<SnapshotConfigs['metadata']>(
    path.join(defaultChainSnapshotDir, 'metadata.json')
  );

  if (envChainMetadata) {
    chainConfig = readJsonFile<SnapshotConfigs['chainConfig']>(
      path.join(defaultChainSnapshotDir, envChainMetadata.chain_config)
    );

    stateConfig = readJsonFile<SnapshotConfigs['stateConfig']>(
      path.join(defaultChainSnapshotDir, envChainMetadata.table_encoding.Json.filepath)
    );
  }
}

function readJsonFile<T>(filePath: string): T | undefined {
  try {
    const fileContent = readFileSync(filePath, 'utf-8');
    return JSON.parse(fileContent) as T;
  } catch (error) {
    // Handle how you see fit
    throw new FuelError(`Failed to read JSON file: ${filePath}`);
    return undefined;
  }
}
```

Reply:
I don't see a reason to handle them in a specific manner, and the error thrown by `readFileSync` is already descriptive enough:
`Error: ENOENT: no such file or directory, open '/tmp/.fuels-ts/d2860af5-db6c-49b0-b315-dbf035bce485/metadata-not-existing.json'`.

`FuelError` should IMO be reserved for domain-specific failure modes (e.g. `ErrorCode.INVALID_MULTICALL`).